Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -9860,6 +9862,120 @@ void WebPage::hasActiveNowPlayingSessionChanged(bool hasActiveNowPlayingSession)
     send(Messages::WebPageProxy::HasActiveNowPlayingSessionChanged(hasActiveNowPlayingSession));
 }
 
+void WebPage::simulateClickOverText(const String& targetText, CompletionHandler<void(bool)>&& completion)
+{
+    RefPtr localMainFrame = m_mainFrame->coreLocalFrame();
+    if (!localMainFrame)
+        return completion(false);
+
+    RefPtr view = localMainFrame->view();
+    if (!view)
+        return completion(false);
+
+    RefPtr document = localMainFrame->document();
+    if (!document)
+        return completion(false);
+
+    RefPtr bodyElement = document->body();
+    if (!bodyElement)
+        return completion(false);
+
+    struct Candidate {
+        Ref<HTMLElement> target;
+        IntPoint location;
+    };
+
+    Vector<Candidate> candidates;
+
+    auto unobscuredContentRect = view->unobscuredContentRect();
+    auto searchRange = makeRangeSelectingNodeContents(*bodyElement);
+    while (is_lt(treeOrder<ComposedTree>(searchRange.start, searchRange.end))) {
+        auto range = findPlainText(searchRange, targetText, {
+            FindOption::CaseInsensitive,
+            FindOption::AtWordStarts,
+            FindOption::TreatMedialCapitalAsWordStart,
+            FindOption::DoNotRevealSelection,
+            FindOption::DoNotSetSelection,
+        });
+
+        if (range.collapsed())
```

Comment:
Should we assert that targetText is not empty?

Reply:
Good point! Since an empty `targetText` is a misuse of the API, I'll also pair a debug assertion here, with an `NSException` in the `WKWebView` method.